spec/unit/application/lookup_spec.rb

```diff
@@ -546,6 +546,103 @@ def run_lookup(lookup)
         allow(lookup.command_line).to receive(:args).and_return(['g'])
         expect(run_lookup(lookup)).to eql("This is G from facts in facts hash")
       end
+
+      describe 'when retrieving given facts' do
+        before do
+          lookup.options[:node] = node
+          allow(lookup.command_line).to receive(:args).and_return(['g'])
+        end
+
+        it 'loads files with yaml extension as yaml on first try' do
+          filename = File.join(tmpdir('lookup_spec_ext'), "facts.yaml")
+          lookup.options[:fact_file] = filename
+          File.open(filename, "w+") { |f| f.write(<<-YAML.unindent) }
+            ---
+            cx: ' G from facts'
+          YAML
```

Comment:
We have a helper method that can shorten things a bit here, see my tests from `configurer_spec.rb`:
https://github.com/puppetlabs/puppet/blob/9cdbc948967e00540afc021a0cfe64db877953b8/spec/unit/configurer_spec.rb#L1323-L1332